Druid/DRInterpreterToCompiler.class.st

```diff
@@ -0,0 +1,188 @@
+"
+Implements a basic interpreter to compiler (it doesn't provide error handling capabilities) using interpreter primitives provided in a class. The interpreter to compiler generates a new class with Druid JITted versions of the primitive methods.
+
+Its basic usage is:
+
+```language=Pharo
+| interpreterToCompiler |
+
+"" Instantiate an interpreter to compiler providing an AbstractInterpreter subclass with primitives ""
+interpreterToCompiler := DRInterpreterToCompiler fromInterpreterClass: DRBasicCogInterpreterArithmeticPrimitives.
+
+"" Generate the Druid intermediate representation and compile methods in a new class ""
+interpreterToCompiler generateIRAndCompileIn: #MyCoolJITCompilerClass
+
+"" Browse the generated class ""
+Smalltalk tools browser openOn: #MyCoolJITCompilerClass.
+```
+
+
+
+"
+Class {
+	#name : #DRInterpreterToCompiler,
+	#superclass : #DRAbstractCompilerBuilder,
+	#instVars : [
+		'memory',
+		'compilerCompilerClass',
+		'compilerCompiler'
+	],
+	#category : #'Druid-CompilerBuilder'
+}
+
+{ #category : #'instance creation' }
+DRInterpreterToCompiler class >> fromInterpreterClass: anInterpreterClass [
+
+	^  self basicNew
+		interpreterClass: anInterpreterClass;
+		yourself
+]
+
+{ #category : #'accessing - compiler' }
+DRInterpreterToCompiler >> compilerCompiler [
+	"Answer a Druid compiler configured for the receiver's interpreter builder"
+
+	^ compilerCompiler
+		ifNil: [ compilerCompiler := 
+			(self compilerCompilerClass forInterpreter: self newInterpreter)
+					compilerClass: self targetClass;
+					yourself ]
+]
+
+{ #category : #'accessing - compiler' }
+DRInterpreterToCompiler >> compilerCompilerClass [
+
+	^ compilerCompilerClass
+		ifNil: [ compilerCompilerClass := self defaultCompilerCompilerClass ]
+]
+
+{ #category : #'accessing - compiler' }
+DRInterpreterToCompiler >> compilerCompilerClass: aClass [
+
+	compilerCompilerClass := aClass
+]
+
+{ #category : #'accessing - compiler' }
+DRInterpreterToCompiler >> defaultCompilerCompilerClass [
+
+	^ DRPrimitiveCompilerCompiler
+]
+
+{ #category : #'accessing - model' }
+DRInterpreterToCompiler >> generateBuildModel [
```

Comment:
Why generate build model? It's not generating anything, the compilation unit is already there